Core/Modules/CANMessageLookUpModule.c

```diff
@@ -40,3 +42,14 @@ void MotorTemperatureCallback(iCommsMessage_t msg)
 {
 	DebugPrint("MotorTemperatureCallback! %d", msg.standardMessageID);
 }
+
+PUBLIC const ICommsMessageInfo* CANMessageLookUpGetInfo(ICommsMessageLookUpIndex id) {
+	for (int i = 0; i < NUMBER_CAN_MESSAGE_IDS; i++) {
+		if (CANMessageLookUpTable[i].index == id) {
+			return &CANMessageLookUpTable[i];
+		}
+	}
+
+	DebugPrint("%s Error! Index [%d] is not defined in lookup!", CLM_TAG, id);
+	return 0;
```

Comment:
If you are returning an error code, I'd prefer you have the function return the result by reference and if the message is not defined, return an error code.